core/src/main/java/org/infinispan/functional/impl/MetaParamsExternalizer.java

```diff
@@ -0,0 +1,39 @@
+package org.infinispan.functional.impl;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Set;
+
+import org.infinispan.commons.api.functional.MetaParam;
+import org.infinispan.commons.marshall.AbstractExternalizer;
+import org.infinispan.commons.util.Util;
+import org.infinispan.marshall.core.Ids;
+
+public final class MetaParamsExternalizer extends AbstractExternalizer<MetaParams> {
```

Comment:
Just returning Iterable would be be fine. The client might expect a subtype but that should work. Don't think I'd want to force this into any array.